### PR TITLE
Porting most things from TGstation (Actions refactoring, statpanel reworking, alt examine reworking, text_ref, defines for Byond 515)

### PR DESCRIPTION
**Pull request by DeadLineSS13**
## About The Pull Request
This PR porting tgstation/tgstation#67083, tgstation/tgstation#69825, tgstation/tgstation#66971 and part porting tgstation/tgstation#70729, tgstation/tgstation#71161

## Why It's Good For The Game
Theses things cool and most refactoring.

## Changelog

:cl: Melbert, LemonInTheDark
add: The Mjolnir wizard loadout no longer scams you. It now also grants 1 level of Tesla Blast!
fix: Spells show up in the stat panel again
fix: Fixes runtime errors that can occur if a wizard spellbook randomizes into unpurchasable spells.
fix: Genetics powers now have their own background, which apparently they have always meant to have.
fix: Fixes some cases where chaplains are unable to cast their own spells.
fix: Fixes some granters being infinite when they should've been limited (Maint loot books).
fix: Wizard's roundend report is now more accurate in what spells they purchased.
qol: Some more spells and actions now use balloon alerts.
qol: Many actions will now more accurately have their icon redded out if you can't use them at the moment.
qol: Genetics mindread is now a pointed ability.
qol: The mimery book now grants you a vow of silence action, like the guide to advanced mimery.
qol: Xenomorph's corrosive acid ability is now a click-to-work action, so you click on what you want to acid.
qol: Also, xenomorph larva now use a radial when choosing an evolution type.
qol: Summon Item now takes a few seconds to remove the mark from an item, so you no longer accidentally remove it.
balance: The mindswap book now gives a slight message warning it's changed on read.
balance: Sacred Flame now lets people around them know they're actually being made super flammable.
balance: Olfaction can't be cast if you have no nose (you're headless).
balance: Being Emotemute now prevents you from invoking spells which require emotes (mime spells).
balance: All wizard spells except Mindswap can't be cast if the wizard is just a brain. Most spells didn't work, anyways.
balance: PAIs and carded AIs can no longer cast spells. Most spells didn't work, anyways.
balance: You can't mindswap into PAIs and brains(?).
admin: Admins can now add spells by name or typepath.
admin: Spell requirements are bitflags now, so you can modify them with the bit-field VV ui. Pretty neat.
code: proc_holders have been removed from existence.
refactor: Wizard, Heretic, Revenant, and Genetics spells are now all action datums.
refactor: Xenomorph, Spider, and Cytology monster abilities are now all action datums.
refactor: Some malf AI and cult abilities are now fully action datums instead of an unholy mix of proc holder and action.
refactor: Refactored how cult's blood target is set.
refactor: Bloodcrawling and jaunting have been completely refactored. Blood crawling has been exorcised from the living level of mobs, and the consumption part of (s)laughter demons is now handled by the action itself.
refactor: The wizard's teleport scroll now actually holds and casts a version of the "teleport" spell, instead of fake casting it.
refactor: Refactored the wizard spellbook a bit.
refactor: Refactored how all item actions are applied and handled.
del: SDQL spells have been removed.
fix: The alt click tab (obj tab?) will no longer cause massive clientside lag (unless you alt click that 2000 obj large tofu pile)
fix: The alt click menu is snappier now, try usin it and let me know if anything blows up yeah?
/:cl:

### CODE REVIEW

**Comment by Spyroshark** [comment]
>balance: All wizard spells except Mindswap can't be cast if the wizard is just a brain.

mmis shooting fireballs is a feature

File: code/controllers/subsystem/statpanel.dm [hunk 1]
```diff
@@ -23,46 +23,41 @@ SUBSYSTEM_DEF(statpanels)
 	if (!resumed)
 		num_fires++
 		var/datum/map_config/cached = SSmapping.next_map_config
-		var/list/global_data = list(
+		global_data = list(
```
Comment (Kapu1178) [inline code/controllers/subsystem/statpanel.dm:26]: put all my stat panel changes back.

**Comment by Kapu1178** [comment]
image
image
Second may or may not be related. Unsure.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/statpanel.dm (hunk 1 above)
Comment (DeadLineSS13) [reply code/controllers/subsystem/statpanel.dm:26]: > put all my stat panel changes back.

Ready

**Comment by DeadLineSS13** [comment]
> image image Second may or may not be related. Unsure.

Fixed first issue. Second issue  mb maded from first issue.

**Comment by Kapu1178** [comment]
image
image
^ Duplicate spells. **Also:** Spell icons spawn ontop of eachother frequently.

**Unsure if this is repo or PR issue:** Cult stun duration is almost non existant

**Comment by Kapu1178** [comment]
yolo

**Pull request merged**